src/control/system/faultdomain_test.go

```diff
@@ -617,6 +619,68 @@ func TestSystem_FaultDomain_MustCreateChild(t *testing.T) {
 	}
 }
 
+func testFaultDomains(num int) []*FaultDomain {
+	emptyFD := MustCreateFaultDomain()
+	fdList := []*FaultDomain{emptyFD}
+	for i := 0; i < num/2; i++ {
+		rack := emptyFD.MustCreateChild(fmt.Sprintf("rack%d", i))
+		fdList = append(fdList, rack)
+	}
+
+	for j := 0; j < num/2; j++ {
+		fdList = append(fdList, emptyFD.MustCreateChild(fmt.Sprintf("pdu%d", j)))
+	}
+	return fdList
+}
+
+func TestSystem_FaultDomain_Uint64(t *testing.T) {
+	// The main thing we care about with these int values is that they
+	// don't (often) collide.
+
+	fdList := testFaultDomains(5000)
+
+	collisions := 0
+	fdInts := make(map[uint64][]string)
+	for _, fd := range fdList {
+		key := fd.Uint64()
+		matching, exists := fdInts[key]
+		if exists {
+			t.Logf("collision(s) for domain %s: %s", fd, matching)
+			collisions++
+		}
+		matching = append(matching, fd.String())
+		fdInts[key] = matching
+	}
+
+	if collisions > 0 {
+		t.Errorf("Collisions: %d", collisions)
```

Comment:
Since neither the names nor hash functions are random, the results should actually be the same every time. But if we do change the hash function or break these methods somehow, we should easily pick it up by running the unit tests.

Reply:
but it won't fail the test? so relies on someone looking for errors in the test console output?

Reply:
Error fails the test, it just doesn't stop it, like fatal does. In this case it's the same.

Reply:
Oh, right. I always forget that `t.Errorf()` is more than just logging.